src/core/task/Task.ts

```diff
@@ -811,7 +812,14 @@ export class Task extends EventEmitter<TaskEvents> implements TaskLike {
 
 			this.apiConversationHistory.push(messageWithTs)
 		} else {
-			const messageWithTs = { ...message, ts: Date.now() }
+			// For user messages, validate and fix tool_result IDs against the previous assistant message
+			// This is a centralized catch-all that handles all tool_use/tool_result ID mismatches
+			const previousAssistantMessage = this.apiConversationHistory
```

Comment:
This is a bit of a heavy-weight way find the previous assistant message. If we can walk backwards through the array starting with the current index then we'd avoid some memory overhead.